plumbum/cli/application.py

```diff
@@ -320,8 +317,39 @@ def _parse_args(self, argv):
                 else:
                     swfuncs[swinfo.func] = SwitchParseInfo(swname, (val,), index)
 
+        # Extracting arguments from environment variables
+        envindex = 0
+        for env, swinfo in self._switches_by_envar.items():
+            envindex-=1
+            envval = os.environ.get(env)
+            if envval is None:
+                continue
+
+            if swinfo.func in swfuncs:
+                continue # skip if overridden by command line arguments
+
+            val = self._handle_argument(envval, swinfo, env)
+            envname = "$%s" % env
+            if swinfo.list:
+                swfuncs[swinfo.func] = SwitchParseInfo(envname, ([val],), envindex)
```

Comment:
it would be useful to add a comment saying you're handling only one value for lists, and perhaps add an assert to catch wrong usage (if applicable)